flash/image/instance_segmentation/data.py

```diff
@@ -12,23 +12,125 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from functools import partial
-from typing import Any, Callable, Dict, Optional, Tuple, Type, Union
+from typing import Any, Callable, Dict, List, Mapping, Optional, Sequence, Tuple, Type, TYPE_CHECKING, Union
 
+import torch
+
+from flash.core.data.auto_dataset import BaseAutoDataset
 from flash.core.data.data_module import DataModule
-from flash.core.data.io.input import DataKeys, InputFormat
+from flash.core.data.io.input import DataKeys, InputFormat, NumpyInput, PathsInput, TensorInput
 from flash.core.data.io.input_transform import InputTransform
 from flash.core.data.io.output_transform import OutputTransform
+from flash.core.data.utils import image_default_loader
 from flash.core.integrations.icevision.data import IceVisionInput
 from flash.core.integrations.icevision.transforms import default_transforms
-from flash.core.utilities.imports import _ICEVISION_AVAILABLE
+from flash.core.utilities.imports import _ICEVISION_AVAILABLE, _KORNIA_AVAILABLE
 from flash.core.utilities.stages import RunningStage
+from flash.image.data import ImageDeserializer, IMG_EXTENSIONS
 
 if _ICEVISION_AVAILABLE:
+    import torchvision.transforms.functional as FT
     from icevision.parsers import COCOMaskParser, Parser, VOCMaskParser
+    from torchvision.datasets.folder import has_file_allowed_extension
+
 else:
     COCOMaskParser = object
     VOCMaskParser = object
     Parser = object
+if _KORNIA_AVAILABLE:
+    import kornia as K
+
+
+class InstanceSegmentationNumpyInput(NumpyInput):
```

Comment:
I think the segmentation numpy input now looks a bit different on master. Not sure how this will play with icevision, I think it will work for prediction only which is fine if that is the intended use case :smiley: